lockc/src/runc.rs

```diff
@@ -380,7 +373,100 @@ impl RuncWatcher {
             }
         }
 
-        Ok(RuncWatcher { fd })
+        Ok(RuncWatcher {
+            bootstrap_rx,
+            ebpf_tx,
+            fd,
+        })
+    }
+
+    async fn add_container(
+        &self,
+        container_id: String,
+        pid: i32,
+        policy_level: container_policy_level,
+    ) -> Result<(), HandleRuncEventError> {
+        let (responder_tx, responder_rx) = oneshot::channel();
+
+        self.ebpf_tx
+            .send(EbpfCommand::AddContainer {
+                container_id,
+                pid,
+                policy_level,
+                responder_tx,
+            })
+            .await?;
+        responder_rx.await??;
+
+        Ok(())
+    }
+
+    fn add_container_sync(
+        &self,
+        container_id: String,
+        pid: i32,
+        policy_level: container_policy_level,
+    ) -> Result<(), HandleRuncEventError> {
+        debug!("adding container {}", container_id);
+
+        Builder::new_current_thread()
+            .build()?
+            .block_on(self.add_container(container_id, pid, policy_level))?;
+
+        Ok(())
+    }
+
+    async fn delete_container(&self, container_id: String) -> Result<(), HandleRuncEventError> {
+        let (responder_tx, responder_rx) = oneshot::channel();
+
+        self.ebpf_tx
+            .send(EbpfCommand::DeleteContainer {
+                container_id,
+                responder_tx,
+            })
+            .await?;
+        responder_rx.await??;
+
+        Ok(())
```

Comment:
same here

Reply:
To reuse the error from previous line, I would need explicitly convert the error returned by the previous line, I would have to explicitly convert it to `HandleRuncEvent` instance. That conversion happens automatically when you handle errors with `?`, but it doesn't happen when you return the other error type, even if it's included in enum.

I had to make it working with

```rust
responder_rx
            .await?
            .map_err(|e| HandleRuncEventError::from(e))
```

That's definitely too much effort. Handling that error only with `?` and returning `Ok(())` is just shorter.